language/move-lang/src/errors/mod.rs

```diff
@@ -117,14 +117,26 @@ fn convert_loc(files: &Files<String>, file_mapping: &FileMapping, loc: Loc) -> (
     (id, Span::new(begin_index, end_index))
 }
 
-fn render_error(files: &Files<String>, file_mapping: &FileMapping, mut error: Error) -> Diagnostic {
-    let mk_lbl = |err: (Loc, String)| -> Label {
+fn render_error(
+    files: &Files<String>,
+    file_mapping: &FileMapping,
+    mut error: Error,
+) -> Diagnostic<FileId> {
+    let mk_secondary_lbl = |err: (Loc, String)| -> Label<FileId> {
+        let (id, span) = convert_loc(files, file_mapping, err.0);
+        Label::<FileId>::secondary(id, span).with_message(err.1)
+    };
+    let mk_primary_lbl = |err: (Loc, String)| -> Label<FileId> {
         let (id, span) = convert_loc(files, file_mapping, err.0);
-        Label::new(id, span, err.1)
+        Label::<FileId>::primary(id, span).with_message(err.1)
     };
+
     let err = error.remove(0);
+    let mut labels = vec![mk_primary_lbl(err)];
+    labels.extend(error.into_iter().map(mk_secondary_lbl));
+
```

Comment:
This is totally fine. But if I had to write it now that there are these two separate labels
```
let mk_lbl = |(i, e)| if i == 0 { mk_primary_lbl(e) } else { mk_secondary_lbl(e) };
Diagnostic::<FileId>::error().with_labels(error.into_iter().map(mk_lbl))
```